docs: add "Overriding recipient connections" for Slack chat templates (#1544)
knocklabs/docs: content/integrations/chat/slack/overriding-recipient-connections.mdx

---
title: Overriding a message's recipient connection
description: Override where a Slack chat message is delivered from the template's settings, instead of using the recipient's stored connections.
tags: ["slack", "chat"]
section: Integrations > Slack
layout: integrations
---

By default, a Slack step delivers to the [channel data](/managing-recipients/setting-channel-data) stored on the recipient. A **recipient connection override** lets a single message template deliver somewhere else — a fixed channel, a specific user, or an incoming webhook — regardless of the recipient's stored connections.

> This is useful when a message should always go to one place: an alerts channel, an internal operations channel, or a destination you compute per recipient with Liquid.

Reviewer: ```suggestion
This is useful when a message should go somewhere other than the recipient's configured destination: an alerts channel, an internal operations channel, or a dynamic destination you compute per-recipient with Liquid.
```
I think worth re-wording this, as I could see someone reaching for this for a fully-dynamic setup (i.e. it's _not_ always going to one place, it just needs to go somewhere other than the recipient's configured connections)

## Setting the override

> In the workflow builder, open the Slack step's template, then open **Template settings** (the gear icon) and set the **Recipient connection** field. It's optional so you can leave it blank to use the recipient's stored connections as usual.

Reviewer: ```suggestion
In the workflow builder, open the Slack step's template, then open **Template settings** (the gear icon) and set the **Recipient connection** field. 
```
I think we can leave this out, since we're framing this as an override.

## What you can specify

The override is a JSON object in one of two shapes.

### Incoming webhook

Post to one fixed Slack incoming webhook URL:

```json
{
  "incoming_webhook": {
    "url": "https://hooks.slack.com/services/T00000000/B00000000/XXXX"
  }
}
```

### Channel or direct message with a bot token

Send to a channel (`channel_id`) or a user's direct message (`user_id`), authenticated with a bot token:

```json
{ "channel_id": "C0123456789", "access_token": "xoxb-your-bot-token" }
```

You can leave out `access_token` and let it come from the [access token stored on the tenant](/integrations/chat/slack/overview#tenant-channel-data-requirements) you trigger the workflow with. In that case, provide only the destination:

```json
{ "channel_id": "C0123456789" }
```

## Using Liquid for dynamic destinations

The value is Liquid-capable and rendered at send time, so you can route each message dynamically from workflow or recipient data:

```json
{ "channel_id": "{{ recipient.slack_channel }}" }
```

<Callout
  type="info"

>   title="Slack only"

Reviewer: +1 on this, should be "Slack only." per style guide
